### PR TITLE
Closes #19

### LINKED ISSUE #19: Reserved vouchers monitoring job
Implementation details:
- single instance job must check existing reserved vouchers once per some configurable period
- if a voucher is in Reserved state for more than some configurable timeout then we need to validate its state
- if there is no linked payment request with a voucher then the voucher must be returned to InStock state
- if there is a lined payment request the PaymentManagement service must be called to validate the status of the payment.
- if the payment is successful then the voucher must be marked as Sold
- if the payment is pending then we need to check if some configurable waiting timeout happened for a voucher payment. if we timeouted for it then the voucher should return to be InStock.
- for other payment statuses we need to return the voucher to InStock state

### PR DESCRIPTION
**Pull request by IliyanIlievPH**

### CODE REVIEW

File: src/MAVN.Service.SmartVouchers.DomainServices/VouchersService.cs [hunk 1]
```diff
@@ -17,9 +17,11 @@
 
 namespace MAVN.Service.SmartVouchers.DomainServices
 {
-    public class VouchersService : IVouchersService
+    public class   VouchersService : IVouchersService
```
Comment (starkmsu) [inline src/MAVN.Service.SmartVouchers.DomainServices/VouchersService.cs]: Why new whitespaces?

File: src/MAVN.Service.SmartVouchers.DomainServices/VouchersService.cs [hunk 2]
```diff
@@ -17,9 +17,11 @@
...
     {
         private const int MaxAttemptsCount = 5;
+        private const string SuccessPaymentStatus = "success";
```
Comment (starkmsu) [inline src/MAVN.Service.SmartVouchers.DomainServices/VouchersService.cs:23]: It's "Success" and "Pending"

File: src/MAVN.Service.SmartVouchers.DomainServices/VouchersService.cs [hunk 3]
```diff
@@ -285,6 +287,40 @@ public Task<VouchersPage> GetCustomerVouchersAsync(Guid customerId, PageInfo pag
...
+
+                switch (paymentManagementResponse.ToLower())
+                {
+                    case SuccessPaymentStatus:
+                        await ProcessPaymentRequestAsync(paymentRequestId.Value);
+                        continue;
+                    case PendingPaymentStatus when voucher.PurchaseDate > DateTime.UtcNow - finishPaymentTimeoutPeriod:
+                        continue;
+                }
+
```
Comment (starkmsu) [inline src/MAVN.Service.SmartVouchers.DomainServices/VouchersService.cs:328]: Need to log received payment status

File: src/MAVN.Service.SmartVouchers.MsSqlRepositories/Repositories/PaymentRequestsRepository.cs [hunk 4]
```diff
@@ -40,5 +41,15 @@ public async Task<string> GetVoucherShortCodeAsync(Guid paymentRequestId)
                 return result?.VoucherShortCode;
             }
         }
+
+        public async Task<Guid?> PaymentRequestExistsAsync(string voucherShortCode)
+        {
+            using (var context = _contextFactory.CreateDataContext())
+            {
+                var result = await context.PaymentRequests.FirstOrDefaultAsync(p => p.VoucherShortCode == voucherShortCode);
```
Comment (starkmsu) [inline src/MAVN.Service.SmartVouchers.MsSqlRepositories/Repositories/PaymentRequestsRepository.cs:49]: Index for voucherShortCode is missing

File: src/MAVN.Service.SmartVouchers.DomainServices/VouchersService.cs (hunk 2 above)
Comment (IliyanIlievPH) [reply src/MAVN.Service.SmartVouchers.DomainServices/VouchersService.cs:23]: I am comparing it with the received status .ToLower() that's why they are with lower case

**Review by starkmsu — APPROVED** [review]

**Pull request merged**